core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggester.java

```diff
@@ -56,7 +59,10 @@ public TermSuggestion innerExecute(String name, TermSuggestionContext suggestion
             );
             Text key = new BytesText(new BytesArray(token.term.bytes()));
             TermSuggestion.Entry resultEntry = new TermSuggestion.Entry(key, token.startOffset, token.endOffset - token.startOffset);
-            for (SuggestWord suggestWord : suggestedWords) {
+            if(suggestion.getDirectSpellCheckerSettings().exactMatch()){
```

Comment:
code formatting: ```if (suggestion.getDirectSpellCheckerSettings().exactMatch()) { ``` instead of ```if(suggestion.getDirectSpellCheckerSettings().exactMatch()){```.